index.js

```diff
@@ -3,53 +3,190 @@ const state = {
     {
       id: "001-beetroot",
       name: "beetroot",
-      price: 0.35
+      price: 0.35,
     },
     {
       id: "002-carrot",
       name: "carrot",
-      price: 0.35
+      price: 0.35,
     },
     {
       id: "003-apple",
       name: "apple",
-      price: 0.35
+      price: 0.35,
     },
     {
       id: "004-apricot",
       name: "apricot",
-      price: 0.35
+      price: 0.35,
     },
     {
       id: "005-avocado",
       name: "avocado",
-      price: 0.35
+      price: 0.35,
     },
     {
       id: "006-bananas",
       name: "bananas",
-      price: 0.35
+      price: 0.35,
     },
     {
       id: "007-bell-pepper",
       name: "bell pepper",
-      price: 0.35
+      price: 0.35,
     },
     {
       id: "008-berry",
       name: "berry",
-      price: 0.35
+      price: 0.35,
     },
     {
       id: "009-blueberry",
       name: "blueberry",
-      price: 0.35
+      price: 0.35,
     },
     {
       id: "010-eggplant",
       name: "eggplant",
-      price: 0.35
-    }
+      price: 0.35,
+    },
   ],
-  cart: []
-};
\ No newline at end of file
+  cart: [],
+};
+//select existing html elements
+// select the store items ul
+const storeItem = document.querySelector(".store--item-list");
+const cartList = document.querySelector(".cart--item-list");
+const totalNum = document.querySelector(".total-number");
+
+function addToCart(item) {
+  // const numAlreadyInCart = state.cart.filter((e) => e.id == item.id).length;
+  // state.cart.push({ quantity: numAlreadyInCart + 1, ...item });
+
+  const existingItems = state.cart.filter((e) => e.id == item.id);
+  // const existingItems = state.cart.filter((e) => e['id'] == item['id']);
+
+  if (existingItems[0]) {
+    existingItems[0].quantity++;
+  } else {
+    state.cart.push({ quantity: 1, ...item });
```

Comment:
I wasn't aware of being able to modify and push an object into an array like this!